src/test/java/com/nvidia/spark/rapids/jni/RmmSparkTest.java

```diff
@@ -774,6 +780,46 @@ public void retryWatchdog() {
     System.err.println("Took " + (endTime - startTime) + "ns to retry 500 times...");
   }
 
+  @Test
+  public void testAllocationDuringSpill() {
```

Comment:
Can we add some more explanation as to why this is possible or we want to be able to deal with it.